pandas/core/generic.py

```diff
@@ -10414,6 +10414,7 @@ def pct_change(self, periods=1, fill_method="pad", limit=None, freq=None, **kwar
             data = self.fillna(method=fill_method, limit=limit, axis=axis)
 
         rs = data.div(data.shift(periods=periods, freq=freq, axis=axis, **kwargs)) - 1
+        rs = rs.loc[~rs.index.duplicated()]
```

Comment:
this is not correct, I think you just need to 
```data.asfreq(freq=freq).div(data.shift(....))```

Reply:
I also like a try at `.asfreq`. However, this `.asfreq` approach is opposite to existing tests in the pandas/tests/series/test_timeseries.py.

https://github.com/pandas-dev/pandas/blob/0436570f05c3b6e7bbb7c7d8fc8fa2f28a0420a8/pandas/tests/series/test_timeseries.py#L394-L403

the above test expects that `freq` and `periods` have the same behavior. I don't know it's ok to do that but, I really don't want to break existing tests.